classes/TaxonomyCleaner.php

```diff
@@ -333,49 +333,45 @@ public function remapOccurrenceTaxon($collid, $oldSciname, $tid, $idQualifier =
 
 			//Get new name and author
 			$newSciname = '';
-			$newAuthor= '';
-			$sql = 'SELECT sciname, author FROM taxa WHERE (tid = '.$tid.')';
+			$newAuthor = '';
+			$newFamily = '';
+			$sql = 'SELECT t.sciname, t.author, ts.family FROM taxa t INNER JOIN taxstatus ts ON t.tid = ts.tid WHERE (t.tid = '.$tid.') AND (ts.taxauthid = 1)';
 			$rs = $this->conn->query($sql);
 			while($r = $rs->fetch_object()){
```

Comment:
I am not a fan of `while($r = $rs->fetch_object()){` because the use of `WHILE` implies multiple results are expected (instead of just one result)  It also causes an additional extraneous condition check.  I would use `if($r = $rs->fetch_object()){`